contracts/src/interfaces/IStakeTable.sol

```diff
@@ -0,0 +1,127 @@
+
+    // === Queuing Stats ===
+
+    /// @notice Get the next available epoch for new registration
+    function nextRegisterationEpoch() external view returns (uint64);
+    /// @notice Get the number of pending registration requests in the waiting queue
+    function numPendingRegistrations() external view returns (uint64);
+    /// @notice Get the next available epoch for exit
+    function nextExitEpoch() external view returns (uint64);
+    /// @notice Get the number of pending exit requests in the waiting queue
+    function numPendingExit() external view returns (uint64);
+
+    // === Write APIs ===
+
+    /// @notice Register a validator in the stake table, transfer of tokens incurred!
+    ///
+    /// @param blsVK The BLS verification key
+    /// @param schnorrVK The Schnorr verification key (as the auxiliary info)
+    /// @param amount The amount to register
+    /// @param stakeType The type of staking (native or restaking)
+    /// @param blsSig The BLS signature that authenticates the `blsVK` field
+    /// @param maxWait The maximum epoch the sender is waiting to wait to be included (cannot be
+    /// smaller than the current epoch)
+    ///
+    /// @return success status
+    ///
+    /// @dev No validity check on `schnorrVK`, as it's assumed to be sender's responsibility,
+    /// the contract only treat it as auxiliary info submitted by `blsVK`.
+    function register(
+        BN254.G1Point calldata blsVK,
+        EdOnBN254.EdOnBN254Point calldata schnorrVK,
+        uint64 amount,
+        StakeType stakeType,
+        bytes calldata blsSig,
+        uint64 maxWait
+    ) external returns (bool);
+
+    /// @notice Deposit more stakes to registered keys
+    ///
+    /// @param blsVK The BLS verification key
+    /// @param amount The amount to deposit
+    /// @return (newBalance, effectiveEpoch) the new balance effective at a future epoch
+    function depoist(BN254.G1Point calldata blsVK, uint64 amount)
+        external
+        returns (uint64, uint64);
+
+    /// @notice Request to exit from the stake table, not immediately withdrawable!
+    ///
+    /// @param blsVK The BLS verification key to exit
+    /// @return success status
+    function exit(BN254.G1Point calldata blsVK) external returns (bool);
+
+    /// @notice Withdraw from the staking pool. Transfers occur! Only successfully exited keys can
+    /// withdraw past their `exitEpoch`.
+    ///
+    /// @param blsVK The BLS verification key to withdraw
+    /// @return The total amount withdrawn, equal to `Node.balance` associated with `blsVK`
+    function withdraw(BN254.G1Point calldata blsVK) external returns (uint64);
+
+    /// @notice Advance to the next epoch, the new `votingStakeTableCommitment` will be
+    /// `frozenStakeTableCommitment` and the new `frozenStakeTableCommitment` will be
+    /// `curEpochComm`,
+    function advanceEpoch(bytes32 curEpochComm) external returns (bool);
```

Comment:
Who calls this function and why?

Reply:
Think of this API as the "setter for stake commitments" (as we have getter as `votingStakeTableCommitment()` and `frozenStakeTableCommitment()`)

So light client prover will submit new states and call this function to update the new finalized stake table commitments.